testsuite/standalone/src/test/java/org/wildfly/extras/creaper/core/online/SslOnlineManagementClientTest.java

```diff
@@ -0,0 +1,157 @@
+ * with Arquillian.</p>
+ *
+ * @see org.wildfly.extras.creaper.core.online.operations.admin.ReloadWhenRestartRequiredTest
+ */
+@RunAsClient
+@Category({ManualTests.class, WildFlyTests.class})
+public class SslOnlineManagementClientTest extends OnlineManagementClientTest {
+
+    private static final String SECURITY_REALM = "SslSecuredTestRealm";
+
+    private static final String SERVER_KEY_ALIAS = "serverKeyAlias";
+    private static final String SERVER_KEY_PASSWORD = "serverKeyPassword";
+    private static final String SERVER_CERTIFICATE_ALIAS = "serverCertificateAlias";
+    private static final String SERVER_CERTIFICATE_PASSWORD = "serverCertificatePassword";
+    private static final String CLIENT_KEY_ALIAS = "clientKeyAlias";
+    private static final String CLIENT_KEY_PASSWORD = "clientKeyPassword";
+    private static final String CLIENT_CERTIFICATE_ALIAS = "clientCertificateAlias";
+    private static final String CLIENT_CERTIFICATE_PASSWORD = "clientCertificatePassword";
+
+    private static final ConfigurationFileBackup CONFIGURATION_BACKUP = new ConfigurationFileBackup();
+    private static OfflineManagementClient offlineClient;
+
+    private static File serverKeyStoreFile;
+    private static File serverTrustStoreFile;
+    private static File clientKeyStoreFile;
+    private static File clientTrustStoreFile;
+    private static SslOptions sslOptions;
+
+
+    @ArquillianResource
+    private ContainerController controller;
+
+
+    @BeforeClass
+    public static void prepareAndConfigure() throws IOException, CommandFailedException, GeneralSecurityException {
+        offlineClient = ManagementClient.offline(OfflineOptions.standalone().rootDirectory(new File("target/jboss-as"))
+                        .configurationFile("standalone.xml").build());
+
+
+        offlineClient.apply(CONFIGURATION_BACKUP.backup());
+
+        prepareKeyAndTrustStoreFiles();
+        prepareSslOptions();
+        configureServerManagementInterfaces();
+    }
+
+    @AfterClass
+    public static void restoreConfiguration() throws CommandFailedException {
+        offlineClient.apply(CONFIGURATION_BACKUP.restore());
+    }
+
+    @AfterClass
+    public static void deleteKeyAndTrustStoreFiles() {
+        serverKeyStoreFile.delete();
+        serverTrustStoreFile.delete();
+        clientKeyStoreFile.delete();
+        clientTrustStoreFile.delete();
+    }
+
+    @Before
+    @Override
+    public void connect() throws IOException {
+        if (this.controller.isStarted(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE)) {
+            client = ManagementClient.online(OnlineOptions.standalone().localDefault().ssl(sslOptions).build());
+        }
+    }
+
+    @Test
+    @InSequence(Integer.MIN_VALUE)
+    public void startServer() throws IOException, CommandFailedException, GeneralSecurityException {
+        this.controller.start(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE);
+    }
+
+    @Test
+    @InSequence(Integer.MAX_VALUE)
```

Comment:
It would be easier to understand. But, It would be also slower -- starting/stopping server, creating/deleting keystores, ...

Is it really so hard to understand that we should make it slower?